### PR TITLE
Add support for GXC address encoding and deconding functionalities

### LINKED ISSUE #196: Add support for GXChain (GXC) addresses in encoding library
Here is the encoding library: https://github.com/ensdomains/address-encoder

Example: https://github.com/ensdomains/address-encoder/pull/47/files

Discord for dev help: https://discord.com/invite/AskZbFx

Please provide a reference link so we can check you did the encoding correctly.

We'd like to keep the encoding library as small as possible, so please be efficient in your use of space.

We will likely be reviewing PRs and paying out bounties more or less on a weekly basis. No need to ping us to check your PR, we will definitely get to it.

Thanks for your help!

### PR DESCRIPTION
**Pull request by janus**
## Issue number
closed #196

## Description
When converted to a string, checksum calculated as the first 4 bytes ripemd160( address ) is appended to the binary address before converting to base58 Encoder string and the result is appended to the prefix, "GXC". 
To Decode , the first 3 bytes must match the prefix, 'GXC' , then removed. And the the rest  is decoded with base58 Decoder. However, the last 4 bytes are for checksum and should be remove. The remaining ripemd160'd , with its first 4 bytes matching the checksum.

## List of features added/change
- gxcAddressEncoder
- gxcAddressDecoder
- getConfig('GXC', 53890, gxcAddressEncoder, gxcAddressDecoder)

## How Has This Been Tested?
Add GXC in the test pool so that decoded address should match it text version

## Checklist:

- [x] My code follows the code style of this project.
- [x] My code implements all the required features.

### CODE REVIEW

File: src/__tests__/index.test.ts [hunk 1]
```diff
@@ -420,6 +420,13 @@ const vectors: Array<TestVector> = [
       { text: '0x67316300f17f063085Ca8bCa4bd3f7a5a3C66275', hex: '67316300f17f063085ca8bca4bd3f7a5a3c66275' },
     ],
   },
+  {
+    name: 'GXC',
+    coinType: 53890,
+    passingVectors: [
+      { text: 'GXC6UKk9URcsCuGxLuRDqEuGzAqDkgKbG8AuWXFXsyzc2r9z7A1kw', hex: '02d085655f8060a79a4b12b14e442b8a554ba867bdadce3c2dc39e1a42a01827c0' },
```
Comment (makoto) [inline src/__tests__/index.test.ts:427]: Can you let me know where did you find the test text/hex and coinType? According to SLIP44, the coin type of GXC is 2303, not 53890 https://github.com/satoshilabs/slips/blob/master/slip-0044.md

**Review by makoto — CHANGES_REQUESTED** [review]
https://github.com/ensdomains/address-encoder/pull/238/files#r511860654

**Comment by janus** [comment]
Thanks. Done.

File: src/index.ts [hunk 2]
```diff
@@ -529,6 +529,39 @@ function icxAddressDecoder(data: string): Buffer {
   }
 }
 
+// Referenced from following
+// https://github.com/gxchain/gxb-core/blob/dev_master/libraries/chain/protocol/address.cpp
+function gxcAddressEncoder(data: Buffer): string {  
+  const RIPEMD160 = require('ripemd160');
```
Comment (makoto) [inline src/index.ts:535]: Please `import` at the top of the file

File: src/index.ts [hunk 3]
```diff
@@ -529,6 +529,39 @@ function icxAddressDecoder(data: string): Buffer {
...
+
+  const checksum = new RIPEMD160().update(data).digest();
+
+  return 'GXC' + bs58EncodeNoCheck(Buffer.concat([data, checksum.slice(0, 4)]));
+}
+
+function gxcAddressDecoder(data: string): Buffer {
+  const RIPEMD160 = require('ripemd160');
```
Comment (makoto) [inline src/index.ts:543]: Please `import` at the top of the file

**Comment by janus** [comment]
I don't understand this failure related to crypto module. It seems it is trying to install it ...
https://travis-ci.com/github/ensdomains/address-encoder/jobs/441897423

**Comment by janus** [comment]
@makoto  Check CI configuration

**Comment by janus** [comment]
@makoto  This current error coming from Travis CI would be needing your help. It worked for me when I use this command `sudo npm i`. I am use  node v14 but I don't think it would be different because I also encountered same with node v10. 


```
npm install

> @ensdomains/address-encoder@0.2.0 prepare /home/travis/build/ensdomains/address-encoder

> npm run build

> @ensdomains/address-encoder@0.2.0 build /home/travis/build/ensdomains/address-encoder

> microbundle --external crypto

(Rollup Core plugin) Error: Could not load stream (imported by /home/travis/build/ensdomains/address-encoder/node_modules/hash-base/index.js): ENOENT: no such file or directory, open 'stream'

Error: Could not load stream (imported by /home/travis/build/ensdomains/address-encoder/node_modules/hash-base/index.js): ENOENT: no such file or directory, open 'stream'

```

**Comment by makoto** [comment]
@janus Please try again. We merged https://github.com/ensdomains/address-encoder/pull/248 which removed dependency of `crypto`

**Comment by janus** [comment]
@makoto  Updated. Please go through.

**Comment by janus** [comment]
@makoto Resolved. Go through and close

File: src/index.ts (hunk 2 above)
Comment (janus) [reply src/index.ts]: Resolved

**Pull request merged**